### PR TITLE
add format param to toArray

### PR DESCRIPTION
**Pull request by glensc**
this allows exporting headers in raw or encoded form

### CODE REVIEW

**Comment by glensc** [comment]
is zend-mail no longer maintained? or why such trivial PR is not responded?

**Comment by froschdesign** [comment]
@glensc 
We are currently working on other components.

In the meantime you can add unit tests.

**Comment by glensc** [comment]
and so are other my tickets just quiet:

  - #69 
  - #64 - this is actually blocker why i can't use this library in my project

**Comment by Maks3w** [comment]
Closed due lack of tests

**Comment by glensc** [comment]
test is included

File: test/HeadersTest.php [hunk 1]
```diff
@@ -309,6 +309,31 @@ public function testCastingToStringReturnsAllMultiHeaderValues()
         $this->assertEquals($expected, $string);
     }
 
+    /**
+     * @test that toArray can take format parameter
+     * @link https://github.com/zendframework/zend-mail/pull/61
+     */
+    public function testToArrayFormat()
```
Comment (Maks3w) [inline test/HeadersTest.php]: This tests holds two scenarios. One method, one scenario.

**Comment by Maks3w** [comment]
I have concerns about the new argument.

I think is best to declare a new method like toArrayRaw, toArrayEncoded and deprecate old method toArray

**Comment by Maks3w** [comment]
Additionally I have doubts about why this feature is required. Could you elaborate why we should merge this?

**Comment by glensc** [comment]
it's needed when attempting to reconstruct mail object because Message() constructor needs input in raw format not to throw exception.

**Comment by glensc** [comment]
i added as parameter as other methods seemed to have similar interface: `public function getFieldValue($format = HeaderInterface::FORMAT_RAW)`, seemed more consistent.

File: test/HeadersTest.php (hunk 1 above)
Comment (Maks3w) [inline test/HeadersTest.php]: Please split this test

Comment (glensc) [inline test/HeadersTest.php]: I was waiting when communication finishes, if the communication ends up whole PR being rejected, i don't see point fixing the tests. so please respond after my replies which changes should be done that you initially mentioned (besides test splitting)

**Comment by ezimuel** [comment]
@Maks3w the PR proposed by @glensc makes sense to me. I'm going to merge it.

**Comment by ezimuel** [comment]
@glensc Thanks for the PR and sorry for the delay!

**Comment by glensc** [comment]
@ezimuel thanks and any pointers how to get this to 2.4 branch?

**Comment by Maks3w** [comment]
This is not a critical bug fix. It won't be backported to 2.4.

**Comment by glensc** [comment]
Yet it the change does not break anything. i need php5.3 support

**Comment by weierophinney** [comment]
@glensc The release process for 2.4 requires a release of the entire framework, which is why we only do so for critical or security fixes.

If you require the functionality in that version, you can also extend the class to add it, and consume your extending class within your own code.

**Comment by glensc** [comment]
@weierophinney extending is not possible usually as hardcoded class names in the framework.

for _example_ i can not override [getFieldValue](https://github.com/zendframework/zend-mail/blob/release-2.5.2/src/Header/GenericHeader.php#L141-L148) because all classes extend from `GenericHeader`, not the class i wrote to extend `GenericHeader`. note this is just example why extending is always not possible to patch functionality.

**Comment by glensc** [comment]
@weierophinney i'd be happy if the change just lands in 2.4 tree, whenever next 2.4 release is actually happening. then the [fork i maintain](https://github.com/glensc/zend-mail/tree/develop-2.4) (for [composer use](https://github.com/eventum/composer/commit/dc42d9ea98be85f53f427f7b7b368dfd20755bc7)) would not diverge from upstream code.

**Pull request merged**